includes/classes/ia.front.wall.php

```diff
@@ -24,14 +24,17 @@
  *
  ******************************************************************************/
 
-class iaWallpost extends abstractModuleFront
+class iaWall extends abstractModuleFront
 {
     protected static $_table = 'wall_posts';
 
     public function insert(array $data)
     {
+        $ip = $this->iaCore->factory('util')->getIp();
+
         if (!$this->iaCore->get('wall_allow_wysiwyg')) {
             $data['body'] = htmlspecialchars($data['body']);
+            $data['ip'] = $ip;
```

Comment:
You always need IP address, no matter if it's WYSIWYG or not.

and you can assign it directly without any interim variable